ui/packages/shared/profile/src/ProfileSelector/index.tsx

```diff
@@ -109,18 +109,6 @@ const ProfileSelector = ({
 
   const [queryExpressionString, setQueryExpressionString] = useState(querySelection.expression);
 
-  useEffect(() => {
-    if (querySelection.expression === undefined) {
-      return;
-    }
-
-    setIsDataLoading(true);
-    setQueryExpression();
-    setIsDataLoading(false);
-
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [timeRangeSelection]);
```

Comment:
this seems like we had an intention behind this, are we sure we're not breaking something else with this?

Reply:
I just updated the PR description about this. Basically, this effect feels like a hack as it watches on the `timeRangeSelection` and resets the query state, which is only necessary as any `timeRangeSelection` would do reset the state by itself. 

For now, the app works fine without this on all the common use cases I tried, but we can watch out for any other edge case issue and see how to properly handle that.